File: src/main/java/com/practice/web/dto/OrderDto.java
package com.practice.web.dto;

import com.practice.theater.models.Order;

import java.math.BigDecimal;
import java.time.Instant;

public class OrderDto {
    private long id;
    private BigDecimal cost;
    private int quantity;
    private UserDto user;
    private boolean confirmed;
    private boolean paid;
    private Instant date;
    private UserDto courier;

    public static OrderDto from(Order order) {
        OrderDto dto = new OrderDto();
        dto.id = order.getId();
        dto.cost = order.getCost();
        dto.quantity = order.getQuantity();
        dto.user = UserDto.from(order.getUser());
        if (order.getCourier() != null) {
            dto.courier = UserDto.from(order.getCourier());
        }
        dto.confirmed = order.isConfirmed();
        dto.paid = order.isPaid();
        dto.date = order.getDate();
        return dto;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public BigDecimal getCost() {
        return cost;
    }

    public void setCost(BigDecimal cost) {
        this.cost = cost;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public UserDto getUser() {
        return user;
    }

    public void setUser(UserDto user) {
        this.user = user;
    }

    public boolean isConfirmed() {
        return confirmed;
    }

    public void setConfirmed(boolean confirmed) {
        this.confirmed = confirmed;
    }

    public boolean isPaid() {
        return paid;
    }

    public void setPaid(boolean paid) {
        this.paid = paid;
    }

    public Instant getDate() {
        return date;
    }

    public void setDate(Instant date) {
        this.date = date;
    }

    public UserDto getCourier() {
        return courier;
    }

    public void setCourier(UserDto courier) {
        this.courier = courier;
    }
}

File: frontend/src/models/dto/OrderDto.js
import {UserDto} from "./UserDto";

export class OrderDto {
    /**
     * @typedef {{
     *  id: Number,
     *  cost: Number,
     *  quantity: Number,
     *  user: UserDto,
     *  courier: UserDto
     *  confirmed: boolean
     *  paid: boolean
     *  date: {nano: Number, epochSecond: Number}
     * }} JavaCompletedOrderDto
     */

    /**
     * @param obj {JavaCompletedOrderDto}
     */
    constructor(obj) {
        this._id = obj.id;
        this._cost = obj.cost;
        this._quantity = obj.quantity;
        this._user = obj.user;
        this._confirmed = obj.confirmed;
        this._courier = obj.courier;
        this._paid = obj.paid;
        this._date = new Date(obj.date.epochSecond * 1000);
    }

    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get cost() {
        return this._cost;
    }

    set cost(value) {
        this._cost = value;
    }

    get quantity() {
        return this._quantity;
    }

    set quantity(value) {
        this._quantity = value;
    }

    get user() {
        return this._user;
    }

    set user(value) {
        this._user = value;
    }

    get confirmed() {
        return this._confirmed;
    }

    set confirmed(value) {
        this._confirmed = value;
    }

    get courier() {
        return this._courier;
    }

    set courier(value) {
        this._courier = value;
    }

    get paid() {
        return this._paid;
    }

    set paid(value) {
        this._paid = value;
    }

    get date() {
        return this._date;
    }

    set date(value) {
        this._date = value;
    }
}